src/Subscriber/PaymentSubscriber.php

```diff
@@ -247,7 +247,7 @@ public function onCheckoutConfirmLoaded(PageLoadedEvent $event)
 
         $page->setPaymentMethods($filteredPaymentMethods);
 
-        $stateDataPaymentMethod = $this->paymentStateDataService->getPaymentMethodType(
+        $stateDataIsStored = (bool) $this->paymentStateDataService->getPaymentStateDataFromContextToken(
```

Comment:
@peterojo just to double check, do we remove the state.data when we select a payment method without stat.data after we've just selected and submitted one method with state.data?

Reply:
Yes we do, see [here](https://github.com/Adyen/adyen-shopware6/blob/61af19439f5670f5c7977ff703242ee551ea97e8/src/Subscriber/PaymentSubscriber.php#L431). When saving a new PM, if the state is empty, we delete any existing state data for the context

Reply:
Thanks!